javascript/react/introduction-to-hooks/effect-hooks.md

```diff
@@ -0,0 +1,127 @@
+---
+author: kapnobatai136
+
+levels:
+
+  - medium
+  
+aspects:
+
+  - new
+
+  - workout
+
+  - deep
+
+type: normal
+
+category: how to
+
+links:
+  - '[effect-hooks](https://reactjs.org/docs/hooks-effect.html){documentation}'
+
+
+---
+
+# Effect hooks
+
+---
+## Content
+
+Effect hooks allow you to perform side effects in functional components. Some examples of side effects include setting up a subscription, manually changing the DOM or data fetching. Let's use the same `<Enki />` component that was created in the previous insight:
+
+```jsx
+import React, {
+  useState,
+  useEffect
+} from "react";
+
+function Enki() {
+  const [name, setName] = useState("Enki");
+
+  useEffect(() => {
+    document.title = `Welcome ${name}.`;
+  });
+
+  return <p>Hello, my name is {name}</p>;
+}
+```
+
+Notice anything new? We have used the `useEffect()` hook to set the document's title to a new custom message that includes the `name` state. If the syntax looks a bit different it is because we are using template literals[1] that allow us to embed expressions. You can think of the `useEffect()` hook as a replacement for `componentDidMount`, `componentDidUpdate`, and `componentWillUnmount` combined.
+
+Note that you have to import the `useEffect` hook from the `'react'` library.
+
+There are two types of effect hooks: with cleanup or without cleanup. In this insight we will focus on the effects without cleanup.
+
+When using class components, the side effects are defined inside the `componentDidMount`, `componentDidUpdate` or `componentWillUnmount` methods, because defining them in the `render` method would be too early (we want side effects to happen **after** React updates the DOM). Here is how our example would look like if we used class components:
```

Comment:
What do you think of?

```suggestion
When using class components, a common pattern for side effects is to define them in the `componentDidMount` method, update them in the `componentDidUpdate` method, and clean them up in the `componentWillUnmount` method.

Here is how our example would look like if we used class components:
```

Note: I removed one sentence because we probably don't always want to have side effects after render